### PR TITLE
feat: update igp docs

### PR DESCRIPTION
**Pull request by paulbalaji**
resolves https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3203 
resolves https://github.com/hyperlane-xyz/v3-docs/issues/51

### CODE REVIEW

File: docs/protocol/interchain-gas-payment.mdx [hunk 1]
```diff
@@ -1,59 +1,67 @@
...
+The gas overhead is set as part of the destination gas configuration, along with the `gasOracle`:
+
+```solidity
+struct DomainGasConfig {
+    IGasOracle gasOracle;
+    uint96 gasOverhead;
+}
+```
+
+This corresponds to the operation cost of processing a message on the destination, irrespective of the message itself.
```
Comment (aroralanuk) [inline docs/protocol/interchain-gas-payment.mdx:53]: this is not quite accurate, you can have different isms for different message types and hence, different gas overheads for the verify function. Like if the transfer amount is >50k or something.

**Review by aroralanuk — APPROVED** [review]

**Review by avious00 — COMMENTED** [review]
LGTM, thanks for adding Destination Gas Config section

File: docs/protocol/interchain-gas-payment.mdx (hunk 1 above)
Comment (paulbalaji) [reply docs/protocol/interchain-gas-payment.mdx:53]: how about something like this?

```suggestion
This corresponds to the operational cost of processing a message on the destination chain.

:::note
As you can configure different ISMs for different message types, you may have different gas overheads for each ISM's `verify` function.
:::
```

Comment (aroralanuk) [reply docs/protocol/interchain-gas-payment.mdx:53]: lovely

File: docs/protocol/interchain-gas-payment.mdx [hunk 2]
```diff
@@ -1,59 +1,73 @@
...
+struct DomainGasConfig {
+    IGasOracle gasOracle;
+    uint96 gasOverhead;
+}
+```
+
+This corresponds to the operational cost of processing a message on the destination chain.
+
+:::note
+You should ensure the `gasOverhead` sufficiently covers the range of ISMs on your destination chain.
```
Comment (nbayindirli) [inline docs/protocol/interchain-gas-payment.mdx:56]: maybe covers the "set" instead of "range" here

Comment (paulbalaji) [reply docs/protocol/interchain-gas-payment.mdx:56]: ah yes set works better here, will try and hit up on my next docs PR

**Pull request merged**